### DISCUSSION TITLE
How to change the color of an element dinamically ?

### DISCUSSION THREAD

**Discussion by Andromelus** [mermaid-js/mermaid#4255] [Q&A] [answered]
I would like to know if it is possible to change the color of an element (a link between two nodes or a node) after the diagram has been generated.

For example, if we take [this diagram](https://mermaid.live/view#pako:eNo9jrFuxCAQRH8FTc1ZxgGDKa5KuqRJuogGHVwO6QyWs5ZysfzvIY4u0ha7o7czs-JUQoTFx-ynC3t-dZkKFXY4HBklSuAY4zz6FCqzusyYA13iGB1sXUM8--VKDi5vFV2m4Ck-hURlhj3762fk8AuVt1s-wdK8xDv0mHyNHO9i3H9e_rrslTgmn99L-UfqCbviC1b0punU0CmtHoSRrek5brBGNp3shTHK9FILNWwc37tB2-ih1VoPRnS_08rtB7nuTFg) as an example:

```
graph LR
toto --> titi
```

This will generate a link like this:

```
<path marker-end="url(#flowchart-pointEnd)" style="fill:none;" class="edge-thickness-normal edge-pattern-solid flowchart-link LS-toto LE-titi" id="L-toto-titi-0" d="M44.86249923706055,19.5L49.02916590372721,19.5C53.19583257039388,19.5,61.52916590372721,19.5,69.86249923706055,19.5C78.19583257039388,19.5,86.52916590372722,19.5,90.69583257039388,19.5L94.86249923706055,19.5"></path>
```

I wish to assign the red color to the link. I tried to:

- `style="fill: red"`
- disable in developer console the fill: none css attribute
- add `fill="red"` to the path element

All without success.

Is what I am trying to achieve possible, and if yes, how can I do it ?

**Comment by huynhicode** [comment, marked as answer]
Hi @Andromelus 

Maybe try `stroke: red`.  Although that doesn't update the arrow color to red.

![Screenshot 2023-03-30 at 10 48 55 AM](https://user-images.githubusercontent.com/35910788/228922106-e481b3c8-97e3-4095-b311-ec5e2c87750f.png)

You can also try `lineColor: red` before you export the diagram. This will update the link and arrow color.

```
themeVariables: {
     lineColor: red
}
```

[Example here](https://mermaid.live/edit#pako:eNpNkEFvgzAMhf8K8jlFDBKgOfSy9dZdNmmHKRePmDZSSFAapHWI_75AVak32997fpZn6LwmkHAOOF6y04dy0Uef7XaHLJpogMFAYUCjk2ZWLssUxAsNpECmUlOPk40K2BP6wmDwx9J11WyeFVnj6NVbH-7OQFrByhbllpSCU_SfN9eBjGEiBtOoMdKbwXTY8BiO6L69f25BzvALsizKvGqbmlecty-cl4LBLY1FnYuq4WVR79t9I4RYGPxtG4o8SUib6MP7_QOdd705wyP6uDGQPdorLf-siVz1)

![Screenshot 2023-03-30 at 10 50 16 AM](https://user-images.githubusercontent.com/35910788/228922322-4c36ce42-60e6-488d-b4f6-9e19bef4c07f.png)

[Documentation here](https://mermaid.js.org/config/theming.html#theme-variables)

**Reply by Andromelus** [reply]
Your first solution is almost perfect (I want to update the diagram **after** generation). I will dig into it. Thank you very much.
